pydantic/types.py

```diff
@@ -1055,3 +1056,74 @@ def validate(cls, value: date) -> date:
                 raise errors.DateNotInTheFutureError()
 
             return value
+
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ NANO TIME TYPE ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+TIME_UNITS = {
+    'ns': 1,
+    'ms': 10 ** 6,
+    's': 10 ** 9,
+    'm': 60 * 10 ** 9,
+    'h': 60 * 60 * 10 ** 9,
+    'd': 24 * 60 * 60 * 10 ** 9
+}
+time_string_re = re.compile(r'\s*(\d+\.?\d*)\s*([a-zA-Z]+)?', re.IGNORECASE)
+
+
+class NanoTime(int):
+    @classmethod
+    def __get_validators__(cls) -> 'CallableGenerator':
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, v: StrIntFloat) -> 'NanoTime':
```

Comment:
I can imagine people might well want to use the string parsing part of this function for `timedelta` parsing, for now could we just extract it into a standalone method so it can be called from validators?